jaxrs-clients/src/test/java/feign/TextDelegateDecoderTest.java

```diff
@@ -78,6 +79,13 @@ public void testUsesStringDecoderWithTextPlain() throws Exception {
     }
 
     @Test
+    public void testCannotReturnStringWithMediaTypeJson() throws Exception {
+        assertThatThrownBy(() -> service.getJsonString("foo"))
```

Comment:
@markelliot we chatted about this 1-2 weeks ago and I claimed that `String` return type doesn't work with media type APPLICATION_JSON. is the behavior different in our internal app server?